### PR TITLE
New chat commands system

### PR DESCRIPTION
**Pull request by joaquin95**
- New commands can be added and removed easier than before. Also while the server is running (needs some fixes)
- Commands have a string for the command itself, a string for their syntax and a execute() method
- Added a function to show pretty debug messages (commands use this to show, for example, errors)

### CODE REVIEW

**Comment by MythicManiac** [comment]
Instead of functions, you should use inheritance now that you created the command class.

So for example you'd want

```CS
class ChatCommand
{
    public ChatCommand(...){ ... }
    
    public abstract void Execute(...);
}

class AdCommand : ChatCommand
{
    public AdCommand(...) : base(...){ ... }

    public override void Execute(...)
    {
        var split = arguments.ToLower().Split(' ');
        float ad;
        if (split.Length < 2)
        {
            // Change these so you don't have to create a new command
            ChatCommand command;
            SendDebugMsgFormatted(DebugMsgType.SYNTAXERROR);
            ChatCommandsDictionary.TryGetValue(split[0], out command);
            SendDebugMsgFormatted(DebugMsgType.SYNTAX, command.syntax); 
        }
        else if (float.TryParse(split[1], out ad))
            game.getPeerInfo(peer).getChampion().getStats().setBonusAdFlat(ad);
    }
}
```

And when you want to execute a new AD command, you do
```CS
var command = new AdCommend(...);
command.Execute();
```

So, then you can create a dictionary like so:
```CS
var dictionary = new Dictionary<string, Chatcommand>(){
    {".ad", new AdCommand(...)}
    ...
};
```

And when you parse the chat message, you can just do
```CS
if(dictionary.ContainsKey(<first part of message here>))
    dictionary[<firs part of message>].Execute();
```

**Comment by MythicManiac** [comment]
Also, use `CamelCase`

File: GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs [hunk 1]
```diff
@@ -11,10 +11,565 @@
...
+                debugMsg.Append(game.getPeerInfo(peer).getChampion().getZ());
+                SendDebugMsgFormatted(DebugMsgType.NORMAL, debugMsg.ToString());
+            }
+        }
+
+        class GoldCommand : ChatCommand
+        {
+            public GoldCommand(string command, string syntax) : base(command, syntax) { }
+
+            public override unsafe void Execute(ENetPeer* peer, Game game, bool hasReceivedArguments, string arguments = "")
```
Comment (MyNick) [inline GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs]: what is unsafe overhere?

File: GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs [hunk 2]
```diff
@@ -11,10 +11,565 @@
...
+            {
+                string commands = "";
+                int count = 0;
+                foreach (KeyValuePair<string, ChatCommand> keyValuePair in ChatCommandsDictionary)
+                {
+                    if (!(keyValuePair.Value.IsHidden || keyValuePair.Value.IsDisabled)) // We could also show disabled commands
+                    {                                  // (with a different color, saying they're disabled, etc.)
+                        count += 1;
+                        commands = commands
+                                   + "<font color =\"#E175FF\"><b>"
```
Comment (MyNick) [inline GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs]: please keep this value in variable with name: "theNameOfThisColor"

File: GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs [hunk 3]
```diff
@@ -31,209 +586,34 @@ public unsafe bool HandlePacket(ENetPeer* peer, byte[] data, Game game)
...
+                    {
+                        SendDebugMsgFormatted(DebugMsgType.ERROR, "<font color =\"#E175FF\"><b>" + split[0] + "</b><font color =\"#AFBF00\"> is disabled.");
                         return true;
+                    }
+                    ChatCommandsDictionary[split[0]].Execute(peer, game, true, message.msg);
+                    return true;
+                }
+                else
+                {
+                    SendDebugMsgFormatted(DebugMsgType.ERROR, "<font color =\"#E175FF\"><b>" + split[0] + "</b><font color =\"#AFBF00\"> is not a valid command.");
```
Comment (MyNick) [inline GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs]: again, please keep the colors with appropriate names

File: GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs [hunk 4]
```diff
@@ -11,10 +11,565 @@
...
+                SendDebugMsgFormatted(DebugMsgType.SYNTAX, this.Syntax);
+            }
+        }
+
+        #region Command classes
+        class AdCommand : ChatCommand
+        {
+            public AdCommand(string command, string syntax) : base(command, syntax) { }
+
+            public override unsafe void Execute(ENetPeer* peer, Game game, bool hasReceivedArguments, string arguments = "")
```
Comment (MyNick) [inline GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs]: for all the "Execute" functions. i think that its better to not have "unsafe" if not needed

Comment (MythicManiac) [inline GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs]: It's probably needed as `ENetPeer*` is a pointer, which are unsafe in C# as far as I know

**Pull request merged**